llvm/lib/Target/AIE/AIEBaseHardwareLoops.cpp

```diff
@@ -173,6 +173,54 @@ char AIEBaseHardwareLoops::ID = 0;
 INITIALIZE_PASS(AIEBaseHardwareLoops, DEBUG_TYPE, AIE_HARDWARE_LOOPS_NAME,
                 false, false)
 
+namespace {
+// Everything becomes much simpler if ZOLs exit to their layout successor
+// We can achieve that by splitting a non-fallthrough exit jump off into
+// a new fallthrough block.
```

Comment:
This isn't a requirement somewhere in our block placement?

Reply:
There's no 'our block placement'. The pass that's called like that only sets block alignment. The standard block placement is complicated, and choses to let a guard fall through to the exit block. Ultimately, two loops may have the same exit block and one of the two will lose.
Note that the jump would start a new region, which would act like a basic block boundary anyway.